pipeline/pipeline.go

```diff
@@ -366,12 +392,14 @@ func (p *Pipeline) finalize(event *Event, notifyInput bool, backEvent bool) {
 		p.totalCommitted.Inc()
 		p.totalSize.Add(int64(event.Size))
 
+		p.outSampleMu.Lock()
 		if len(p.outSample) == 0 && rand.Int()&1 == 1 {
 			p.outSample = event.Root.Encode(p.outSample)
 		}
+		p.outSampleMu.Unlock()
 
-		if event.Size > p.maxSize {
-			p.maxSize = event.Size
+		if event.Size > int(p.maxSize.Load()) {
```

Comment:
is it possible to have an overflow here?

Reply:
It's okay unless event msg <2GiB, which is nonsense.